src/Components/Web/src/Forms/InputDate.cs

```diff
@@ -30,12 +35,31 @@ public class InputDate<TValue> : InputBase<TValue>
         /// </summary>
         [DisallowNull] public ElementReference? Element { get; protected set; }
 
+        /// <inheritdoc />
+        public override Task SetParametersAsync(ParameterView parameters)
+        {
+            _typeAttributeValue = parameters.TryGetValue<string>("type", out var typeAttributeValue)
+                ? typeAttributeValue
+                : "date";
```

Comment:
What do you think about using `"time"` by default when `typeof(TValue) == typeof(TimeOnly)`?

Reply:
The design here is based on the developer specifying their chosen `type` attribute, or defaulting to `"date"`. In most cases it's orthogonal to the type of the .NET property being bound, because we convert automatically between all the different date/time .NET types (even if that involves discarding some information).

I would rather keep the rules simple, as they are in this PR. That is, the developer picks a `type` to control the UI appearance, and the associated bound .NET type is independent of that.